experiment/selinux/src/selinux.rs

```diff
@@ -44,268 +122,292 @@ impl SELinux {
-    }
-
-    // function similar with setKeyLabel in go-selinux repo.
-    // set_key_label takes a process label and tells the kernel to assign the
-    // label to the next kernel keyring that gets created.
-    pub fn set_key_label(label: &str) -> Result<(), SELinuxError> {
-        match Self::write_con(Path::new("/proc/self/attr/keycreate"), label) {
-            Ok(v) => Ok(v),
-            // TODO: This line will be fixed after implementing write_con.
-            Err(e) => Err(e),
+    // find_selinux_fs finds the SELinux filesystem mount point.
+    fn find_selinux_fs() -> Option<PathBuf> {
+        // fast path: check the default mount first
+        let selinux_fs_mount_path = PathBuf::from(SELINUX_FS_MOUNT);
+        if Self::verify_selinux_fs_mount(selinux_fs_mount_path.clone()) {
+            return Some(selinux_fs_mount_path);
         }
-    }
-
-    // function similar with KeyLabel in go-selinux repo.
-    // key_label retrieves the current kernel keyring label setting
-    pub fn key_label() {
-        unimplemented!("not implemented yet")
-    }
 
-    // function similar with clearLabels in go-selinux repo.
-    // clear_labels clears all reserved labels.
-    pub fn clear_labels() {
-        unimplemented!("not implemented yet")
-    }
-
-    // function similar with reserveLabel in go-selinux repo.
-    // reserve_label reserves the MLS/MCS level component of the specified label
-    pub fn reserve_label(label: &str) {
-        unimplemented!("not implemented yet")
-    }
-
-    // function similar with roFileLabel in go-selinux repo.
-    // ro_file_label returns the specified SELinux readonly file label
-    pub fn ro_file_label() {
-        unimplemented!("not implemented yet")
-    }
-
-    // function similar with kvmContainerLabels in go-selinux repo.
-    // kvm_container_labels returns the default processLabel and mountLabel to be used
-    // for kvm containers by the calling process.
-    pub fn kvm_container_labels() {
-        unimplemented!("not implemented yet")
-    }
-
-    // function similar with initContainerLabels in go-selinux repo.
-    // init_container_labels returns the default processLabel and file labels to be
-    // used for containers running an init system like systemd by the calling process.
-    pub fn init_container_labels() {
-        unimplemented!("not implemented yet")
-    }
+        // check if selinuxfs is available before going the slow path
+        let fs = fs::read_to_string("/proc/filesystems").unwrap_or_default();
+        if !fs.contains("\tselinuxfs\n") {
+            return None;
+        }
 
-    // function similar with containerLabels in go-selinux repo.
-    // container_labels returns an allocated processLabel and fileLabel to be used for
-    // container labeling by the calling process.
-    pub fn container_labels() {
-        unimplemented!("not implemented yet")
+        // continue reading until finding mount point
+        loop {
```

Comment:
Please tell me the case where we need the loop.

Reply:
I had thought loop is needed because go-selinux has it as well.
But I understood that go-seliux is not infinite loop and my implementation is wrong. I will fix it.

https://github.com/opencontainers/selinux/blob/bb1ec25c889f29e35dffec33346af3c7d434dbc1/go-selinux/selinux_linux.go#L173-L182